app/src/main/java/com/nervousfish/nervousfish/data_objects/RSAKey.java

```diff
@@ -38,15 +38,13 @@ public RSAKey(final String modulus, final String exponent) {
      */
     static public IKey fromJSON(final JsonReader reader) throws IOException {
         final Map<String, String> map = new ConcurrentHashMap<>();
-        reader.beginObject();
         while (reader.hasNext()) {
             final String name = reader.nextName();
             final String value = reader.nextString();
             map.put(name, value);
         }
-        reader.endObject();
 
-        final String modulus = map.get("modules");
+        final String modulus = map.get("modulus");
```

Comment:
Could you make a constant from modulus and exponent, so that you're sure that the read method reads the same key as the write method writes?